### PR TITLE
Fineract-1411 Allow charges to map to all types of Ledgers

### PR DESCRIPTION
**Pull request by ankita10r**
## Description
Describe the changes made and why they were made instead of how they were made.

## Related issues and discussion
#{Issue Number}

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Validate the JS and HTML files with `grunt validate` to detect errors and potential problems in JavaScript code.

- [x] Run the tests by opening `test/SpecRunner.html` in the browser to make sure you didn't break anything.

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `community-app/Contributing.md`.

### CODE REVIEW

File: app/scripts/controllers/product/EditChargeController.js [hunk 1]
```diff
@@ -52,8 +72,11 @@
                 };
 
                 if(data.incomeOrLiabilityAccount){
-                    scope.formData.incomeAccountId = data.incomeOrLiabilityAccount.id;
+                    console.log(data.incomeAndLiabilityAccount);
```
Comment (avikganguly01) [inline app/scripts/controllers/product/EditChargeController.js]: Remove unnecessary log statements.

File: app/scripts/controllers/product/CreateChargeController.js [hunk 2]
```diff
@@ -21,6 +21,28 @@
...
+                var accountMappingForChargeConfigVar = scope.accountMappingForChargeConfig.toLowerCase();
+
+                if(accountMappingForChargeConfigVar.indexOf("asset") > -1){
+                    scope.accountMappingForCharge = scope.accountMappingForCharge.concat(scope.assetAccountOptions);
+                }
+                if(accountMappingForChargeConfigVar.indexOf("liability") > -1){
+                    scope.accountMappingForCharge = scope.accountMappingForCharge.concat(scope.liabilityAccountOptions);
+                }
+               if(accountMappingForChargeConfigVar.indexOf("expense") > -1){
+                scope.accountMappingForCharge = scope.accountMappingForCharge.concat(scope.expenseAccountOptions);
```
Comment (avikganguly01) [inline app/scripts/controllers/product/CreateChargeController.js]: Format the tab indentations please.

File: app/scripts/controllers/product/CreateChargeController.js [hunk 3]
```diff
@@ -21,6 +21,28 @@
...
+                }
+               if(accountMappingForChargeConfigVar.indexOf("income") > -1){
+                scope.accountMappingForCharge = scope.accountMappingForCharge.concat(scope.incomeAccountOptions);
```
Comment (avikganguly01) [inline app/scripts/controllers/product/CreateChargeController.js]: Format the tab indentations please.

**Review by avikganguly01 — APPROVED** [review]

**Comment by avikganguly01** [comment]
@ankita10r : Please squash your commits.

**Comment by ankita10r** [comment]
@avikganguly01 done

**Pull request merged**